Mervebolat/frontend activity section (#37815)
DataDog/documentation: content/en/tracing/services/service_page.md

{{< img src="tracing/visualization/service/costs_tab_1.png" alt="Costs" style="width:90%;">}}

### Frontend activity

{{< callout url="https://www.datadoghq.com/product-preview/journey-monitoring/" btn_hidden="false" header="Join the Preview!">}}

> Join Operations/Journey monitoring preview to start using Frontend activity section!

Reviewer: ```suggestion
Join the Operations/Journey monitoring Preview to start using Frontend activity section!
```

> Join the Operations/Journey monitoring Preview to start using Frontend activity section!

Reviewer 2: I removed "Operations/" because it looks like the [preview link](https://www.datadoghq.com/product-preview/journey-monitoring/?_gl=1*lugjfd*_gcl_au*MTQyNTcwMTMxOS4xNzgxMDE1NDQw*_ga*MjEyNjc5NTM1Ni4xNzgyODI0OTY2*_ga_KN80RDFSQK*czE3ODM0NDY1NjUkbzE4MyRnMSR0MTc4MzQ0NzAyNiRqNjAkbDAkaDQzMTMxMDEz*_fplc*aEliU25KdGRaZURYayUyRmpYTmpHT1pjdVRNZGE1Wk9TUmNMJTJCQTlmUyUyRkx4eE1XM0dZS2ZvJTJCYzkwZEklMkZlWFFVMFhCQyUyRnZsJTJGJTJGamVTTzAycUNHdTk0NEtRc3JsdyUyRnd0MHRpVkdRdzJibEtHaEMlMkJNTFZ2UWlJVkgxWU9VcmpQWVElM0QlM0Q.) doesn't include it. also capitalized Monitoring
```suggestion
Join the Journey Monitoring Preview to start using the Frontend activity section!
```

Author: Actually RUM team has two separate preview program for Operations and Journey monitoring. Added two callouts for each of them.

{{< /callout >}}

> The Frontend activity tab connects your backend service to the Real User Monitoring (RUM) journeys, operations, and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.

Reviewer 2: could we link "Real User Monitoring (RUM)" to the /real_user_monitoring/ page?

> The Frontend activity tab connects your backend service to the Real User Monitoring (RUM) journeys, operations, and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.

Reviewer 2: also, "Frontend activity" should be enclosed in {{< ui >}}...{{< /ui >}} tags in all instances :-) this will bold that text!

Author: Ah, I think we didn't follow this for all the other sections on this page, so for the consistency, I am leaving this as is? But feel free to change if this is the best practice!

> {{< img src="static/images/tracing/services/service_page/front-end-activity.png" alt="FE-activity" style="width:90%;">}}

Reviewer: ```suggestion
{{< img src="tracing/services/service_page/front-end-activity.png" alt="The front end activity side panel showing service, journey, operation, usage details, and related session replays" style="width:90%;">}}
```

> {{< img src="tracing/services/service_page/front-end-activity.png" alt="The front end activity side panel showing service, journey, operation, usage details, and related session replays" style="width:90%;">}}

Reviewer 2: the image is currently broken in the Preview—I'm thinking the path needs a slash 
```suggestion
{{< img src="/tracing/services/service_page/front-end-activity.png" alt="The frontend activity side panel showing service, journey, operation, usage details, and related session replays" style="width:90%;">}}
```

The metrics in this tab are computed from indexed spans that are correlated with RUM data. The header shows how many indexed spans the view is based on, along with the share of total spans that are indexed.

> **Prerequisites**

Reviewer 2: ```suggestion
#### Prerequisites
```

- RUM is set up for your frontend application.
- RUM and APM are connected, so that frontend events are correlated with backend traces from this service.
- RUM Journey and Operations are enabled for your organization.

To populate the Users and Accounts metrics, your RUM SDKs must send the `usr.id` and `account.id` attributes. Without these attributes, the journey, operation, and view metrics are still available, but Datadog cannot attribute activity to specific users or accounts.

> **Filtering the view**

Reviewer 2: ```suggestion
#### Filtering the view
```

Use the cascading selectors at the top of the tab to scope the data, moving from the broadest grouping to the most specific:

> 

Reviewer 2: ```suggestion
```

- Journey: a sequence of steps a user takes in your application, defined in Journey Monitoring.
- Operation: a user operation tracked within a journey.

> - View: an individual RUM view (page or screen) associated with the selected operations.

Reviewer 2: ```suggestion
- Journey: a sequence of steps a user takes in your application, defined in Journey Monitoring
- Operation: a user operation tracked within a journey
- View: an individual RUM view (page or screen) associated with the selected operations
```

> **Summary metrics**

Reviewer 2: ```suggestion
#### Summary metrics
```

The summary cards highlight the aggregate frontend impact for the current selection. Each card shows the value over the selected time frame and the change relative to the previous period:

- Users: number of distinct users (`usr.id`) active in the selected journeys.
- Accounts: number of distinct accounts (`account.id`) active in the selected journeys.
- Requests: number of frontend requests correlated with this service.
- Avg Conversion: average conversion rate across the selected journeys.

> - P95 Latency: 95th percentile latency experienced by the frontend.

Reviewer 2: ```suggestion
- p95 Latency: 95th percentile latency experienced by the frontend.
```
this page's "Columns" section uses lowercase Ps!

- Error Rate: percentage of frontend requests that resulted in an error.

> **Product Analytics enrichment**

Reviewer 2: ```suggestion
#### Product Analytics enrichment
```

> If you use Product Analytics, the Frontend activity tab shows enriched details about the impacted users and accounts, giving you more context on who is affected when the service degrades and helping you prioritize work by business impact.

Reviewer 2: could we link "Product Analytics" to those docs? /product_analytics/

{{< img src="tracing/visualization/service/costs_tab_1.png" alt="Costs" style="width:90%;">}}

### Frontend activity

{{< callout url="https://www.datadoghq.com/product-preview/journey-monitoring/" btn_hidden="false" header="Join the Preview!">}}

> Join the Journey monitoring Preview to start using Frontend activity section!

Reviewer 2: ```suggestion
Join the Journey Monitoring Preview to start using the Frontend activity section!
```

{{< /callout >}}

{{< callout url="https://www.datadoghq.com/product-preview/operations-monitoring/" btn_hidden="false" header="Join the Preview!">}}

> Join the Operations monitoring Preview to start using Frontend activity section!

Reviewer 2: ```suggestion
Join the Operations Monitoring Preview to start using the Frontend activity section!
```

{{< /callout >}}

> The Frontend activity tab connects your backend service to the [Real User Monitoring (RUM)][24] journeys, operations, and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.

Reviewer 2: ```suggestion
The Frontend activity tab connects your backend service to the [Real User Monitoring (RUM)][25] journeys, operations, and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.
```

#### Filtering the view

Use the cascading selectors at the top of the tab to scope the data, moving from the broadest grouping to the most specific:

- Journey: a sequence of steps a user takes in your application, defined in Journey Monitoring
- Operation: a user operation tracked within a journey
- View: an individual RUM view (page or screen) associated with the selected operations

#### Summary metrics

The summary cards highlight the aggregate frontend impact for the current selection. Each card shows the value over the selected time frame and the change relative to the previous period:

- Users: number of distinct users (`usr.id`) active in the selected journeys.
- Accounts: number of distinct accounts (`account.id`) active in the selected journeys.
- Requests: number of frontend requests correlated with this service.
- Avg Conversion: average conversion rate across the selected journeys.
- p95 Latency: 95th percentile latency experienced by the frontend.
- Error Rate: percentage of frontend requests that resulted in an error.

#### Product Analytics enrichment

> If you use [Product Analytics][25], the Frontend activity tab shows enriched details about the impacted users and accounts, giving you more context on who is affected when the service degrades and helping you prioritize work by business impact.

Reviewer 2: ```suggestion
If you use [Product Analytics][26], the Frontend activity tab shows enriched details about the impacted users and accounts, giving you more context on who is affected when the service degrades and helping you prioritize work by business impact.
```

{{< img src="tracing/visualization/service/costs_tab_1.png" alt="Costs" style="width:90%;">}}

### Frontend activity

{{< callout url="https://www.datadoghq.com/product-preview/journey-monitoring/" btn_hidden="false" header="Join the Preview!">}}
Join Operations/Journey monitoring preview to start using Frontend activity section!
{{< /callout >}}

> The Frontend activity tab connects your backend service to the Real User Monitoring (RUM) journeys, operations and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.

Reviewer: ```suggestion
The Frontend activity tab connects your backend service to the Real User Monitoring (RUM) journeys, operations, and views that depend on it. Use it to understand how your service's performance affects the end-user experience and business outcomes, such as which user journeys slow down or fail to convert when the service degrades.
```

{{< img src="static/images/tracing/services/service_page/front-end-activity.png" alt="FE-activity" style="width:90%;">}}

The metrics in this tab are computed from indexed spans that are correlated with RUM data. The header shows how many indexed spans the view is based on, along with the share of total spans that are indexed.

**Prerequisites**
- RUM is set up for your frontend application.
- RUM and APM are connected, so that frontend events are correlated with backend traces from this service.
- RUM Journey and Operations are enabled for your organization.

> To populate the Users and Accounts metrics, your RUM SDKs must send the usr.id and account.id attributes. Without these attributes, the journey, operation, and view metrics are still available, but Datadog cannot attribute activity to specific users or accounts.

Reviewer: ```suggestion
To populate the Users and Accounts metrics, your RUM SDKs must send the `usr.id` and `account.id` attributes. Without these attributes, the journey, operation, and view metrics are still available, but Datadog cannot attribute activity to specific users or accounts.
```

**Filtering the view**

Use the cascading selectors at the top of the tab to scope the data, moving from the broadest grouping to the most specific:

- Journey: a sequence of steps a user takes in your application, defined in Journey Monitoring.
- Operation: a user operation tracked within a journey.
- View: an individual RUM view (page or screen) associated with the selected operations.

**Summary metrics**

The summary cards highlight the aggregate frontend impact for the current selection. Each card shows the value over the selected time frame and the change relative to the previous period:

- Users: number of distinct users (usr.id) active in the selected journeys.

> - Accounts: number of distinct accounts (account.id) active in the selected journeys.

Reviewer: ```suggestion
- Users: number of distinct users (`usr.id`) active in the selected journeys.
- Accounts: number of distinct accounts (`account.id`) active in the selected journeys.
```

- Requests: number of frontend requests correlated with this service.
- Avg Conversion: average conversion rate across the selected journeys.
- P95 Latency: 95th percentile latency experienced by the frontend.
- Error Rate: percentage of frontend requests that resulted in an error.

**Product Analytics enrichment**

> If you use Product Analytics, the Frontend activity tab surfaces enriched details about the impacted users and accounts, giving you more context on who is affected when the service degrades and helping you prioritize work by business impact.

Reviewer: ```suggestion
If you use Product Analytics, the Frontend activity tab shows enriched details about the impacted users and accounts, giving you more context on who is affected when the service degrades and helping you prioritize work by business impact.
```

> ContentFor AI agents: A markdown version of this page is available at https://docs.datadoghq.com/tracing/services/service_page.md. A documentation index is available at /llms.txt.

Reviewer: ```suggestion
Content for AI agents: A markdown version of this page is available at https://docs.datadoghq.com/tracing/services/service_page.md. A documentation index is available at `/llms.txt`.
```

Product
Customers
Pricing
Solutions

> DataDog

Reviewer: ```suggestion
Datadog
```

About
Blog
Login
Datadog Docs

Search documentation...

Essentials

> Getting Startedpasted

Reviewer: Is this content supposed to be here? I'm not quite sure what it is for

Author: This and "...AI agents" part: It was already there when I clicked edit button so didn't want to touch them at first. Now removed all these.

Reviewer: Oh huh, that's odd. Thanks for removing it.
